app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,34 @@
 class PostsController < ApplicationController
   def index
-    @posts = Post.all
+    @user = User.find(params[:user_id])
+    @posts = @user.posts.includes(:comments)
   end
 
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author_id = current_user.id
+
+    if @post.save
+      @post.update_user_post_counter(params[:user_id])
+      redirect_to user_post_path(@user.id)
+      flash[:notice] = 'Your comment was successfully created'
+    else
+      redirect_to new_user_post_path(@user.id)
+      flash[:notice] = 'An error has occurred while creating the post'
+    end
+  end
+
+  private
+
+  def post_params
+    params.require(:post).permit(:tittle, :text, :comments_counter, :likes_counter)
+  end
```

Comment:
- [ ] Kindly only permit input params, in this case for post you only allow `:title` and `:text`. 
`comments_counter` and `likes_counter` suppose to be auto-updated through the post model.